feature/rds/auth/connect.go

```diff
@@ -94,3 +100,26 @@ func BuildAuthToken(ctx context.Context, endpoint, region, dbUser string, creds
 
 	return url, nil
 }
+
+func validateURL(hostPort string) (host, port string) {
+	colon := strings.LastIndexByte(hostPort, ':')
+	if colon != -1 {
+		host, port = hostPort[:colon], hostPort[colon+1:]
+	}
+	if !validatePort(port) {
+		port = ""
+		return
+	}
+	if strings.HasPrefix(host, "[") && strings.HasSuffix(host, "]") {
+		host = host[1 : len(host)-1]
+	}
```

Comment:
Hmmm, this assumes that the position of `[` and `]` is the first and last character.